src/foam/nanos/crunch/ui/CapabilityView.js

```diff
@@ -0,0 +1,172 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.crunch.ui',
+  name: 'CapabilityView',
+  extends: 'foam.u2.View',
+  documentation: 'All purpose capability view that takes in an array of capability ids and displays those capabilities',
+
+  imports: [
+    'crunchController',
+    'notify',
+    'subject',
+    'userCapabilityJunctionDAO'
+  ],
+
+  requires: [
+    'foam.log.LogLevel',
+    'foam.nanos.crunch.UserCapabilityJunction',
+    'foam.u2.detail.SectionView'
+  ],
+
+  constants: [
+    {
+      name: 'MAX_DEPTH',
+      type: 'Integer',
+      value: 5,
+      documentation: 'max. depth addListeners can move down on recursion'
+    }
+
+  ],
+
+  properties: [
+    {
+      class: 'StringArray',
+      name: 'capabilityIDs'
+    },
+    {
```

Comment:
I would really recommend using definitions of variable types in these properties.

capabilityIDs - no problems
capabilities - should be class: 'FObjectArray', of: 'Capabilities',
wizardlets - 'FObjectArray',of: 'BaseWizardlet',
wizardletSectionsList -  class: 'FObjectArray', of: 'foam.layout.Section',
showTitle - class: 'String'

Reply:
wizardletSectionsList is an array of arrays so I had to give 'Array' class.